src/platform/common/application/workspace.ts

```diff
@@ -25,9 +25,14 @@ export class WorkspaceService implements IWorkspaceService {
         return workspace.onDidChangeConfiguration;
     }
     public get rootPath(): string | undefined {
-        return Array.isArray(workspace.workspaceFolders) && workspace.workspaceFolders.length > 0
-            ? workspace.workspaceFolders[0].uri.fsPath
-            : undefined;
+        const firstWorkspace =
+            Array.isArray(workspace.workspaceFolders) && workspace.workspaceFolders.length > 0
+                ? workspace.workspaceFolders[0]
+                : undefined;
+        if (firstWorkspace) {
+            // eslint-disable-next-line local-rules/dont-use-fspath
+            return getOSType() === OSType.Web ? firstWorkspace.uri.path : firstWorkspace.uri.fsPath;
```

Comment:
Special cased this linter rule here. fsPath is no longer allowed in anything except .node.ts files but for this 'rootPath', I figured it made sense to still use it.

Reply:
I think it would be good to add a debt item to ensure we use Uris instead of strings.
If `rootPath` were to reutn the Uri, then we woudln't have to do any of this, I believe that's what @roblourens also suggested, & I can see that saving us from writing this if in a number of places.

Reply:
Will do